index.js

```diff
@@ -55,6 +58,77 @@ app.get("/favicon.ico", (_req, res) => {
 app.all("/graphql", (req, res) => res.redirect("/"))
 
 app.use(bodyParser.json())
+
+function parse_args() {
+  return "( ... )"
+}
+
+function trace(res, span) {
+  span.addTags({
+    "http.status_code": res.statusCode,
+  })
+  span.finish()
+}
+
+app.use((req, res, next) => {
+  if (req.method === "POST") {
+    const tracer = new Tracer({ service: "metaphysics" })
+    const span = tracer.startSpan("metaphysics.query")
+    const query = req.body.query.replace(/(\()([^\)]*)(\))/g, parse_args)
+    span.addTags({
+      resource: query,
+      type: "web",
+      "span.kind": "server",
+      "http.method": req.method,
+      "http.url": req.url,
+    })
+
+    assign(req, { span })
+
+    res.on("finish", () => trace(res, span))
+    res.on("close", () => trace(res, span))
+  }
+  next()
+})
+
+function wrapResolve(typeName, fieldName, resolver) {
```

Comment:
So just to be clear, the entire MP request is a span, and then each of the resolvers will be span under that original one? Or do they ever nest?

Reply:
They do nest. In the wrapper I set the parent span on the request and then call the resolver so if within that call there is another resolver it's parent will be the outer one.